### PR TITLE
fix(ios)(8_0_X): Instead of native exception send javascript exception

### PR DESCRIPTION
**Pull request by vijaysingh-axway**
https://jira.appcelerator.org/browse/TIMOB-27112

### CODE REVIEW

File: iphone/TitaniumKit/TitaniumKit/Sources/Kroll/KrollContext.m [hunk 1]
```diff
@@ -148,7 +148,13 @@ static JSValueRef CommonJSRequireCallback(JSContextRef jsContext, JSObjectRef js
     return [KrollObject toValue:ctx value:result];
   }
   @catch (NSException *e) {
-    return ThrowException(jsContext, [e reason], exception);
```
Comment (garymathews) [inline iphone/TitaniumKit/TitaniumKit/Sources/Kroll/KrollContext.m]: I think the `ThrowException` method should be modified to report the exception. So all exceptions are logged.

```Obj-C
JSValueRef ThrowException(JSContextRef contextRef, NSString *message, JSValueRef *exception)
{
  JSGlobalContextRef globalContextRef = JSContextGetGlobalContext(contextRef);
  JSContext *context = [JSContext contextWithJSGlobalContextRef:globalContextRef];
  JSValue *errorValue = [JSValue valueWithNewErrorFromMessage:message inContext:context];
  [[TiExceptionHandler defaultExceptionHandler] reportScriptError:[TiUtils scriptErrorValue: @{
    @"stack": [errorValue valueForProperty:@"stack"],
    @"message": [errorValue valueForProperty:@"message"]
  }]];

  JSStringRef jsString = JSStringCreateWithCFString((CFStringRef)message);
  *exception = JSValueMakeString(contextRef, jsString);
  JSStringRelease(jsString);
  return JSValueMakeUndefined(contextRef);
}
```

**Review by garymathews — CHANGES_REQUESTED** [review]

File: iphone/TitaniumKit/TitaniumKit/Sources/Kroll/KrollContext.m (hunk 1 above)
Comment (garymathews) [reply iphone/TitaniumKit/TitaniumKit/Sources/Kroll/KrollContext.m]: Actually, maybe you could just do this:
```Obj-C
JSValueRef ThrowException(JSContextRef contextRef, NSString *message, JSValueRef *exception)
{
  JSGlobalContextRef globalContextRef = JSContextGetGlobalContext(contextRef);
  JSContext *context = [JSContext contextWithJSGlobalContextRef:globalContextRef];
  *exception = [JSValue valueWithNewErrorFromMessage:message inContext:context];
  return JSValueMakeUndefined(contextRef);
}
```
I haven't tested this, just an idea.

Comment (vijaysingh-axway) [reply iphone/TitaniumKit/TitaniumKit/Sources/Kroll/KrollContext.m]: Yes. Good idea. Updated PR.

**Review by garymathews — APPROVED** [review]
CR: **PASS**

**Comment by keerthi1032** [comment]
FR Passed. Javascript stake trace is now showing  .
**Test Environment:** 
Operating System
  Name                        = Mac OS X
  Version                     = 10.13.6
  Architecture                = 64bit
Node.js
  Node.js Version             = 8.9.1
  npm Version                 = 5.5.1
Titanium CLI
  CLI Version                 = 5.1.1
Titanium SDK
  SDK Version                 = local sdk 8.1.0.v20190531095040 and local sdk 8.0.2.v20190531094628
Studio = 5.1.2.201903111843
CLI= 7.0.11
Device =iPhone 6s iOS 12
Simulator =iPhone 6 iOS 11

**Pull request merged**